### DISCUSSION TITLE
Can you pass parameters to a repeater?

### DISCUSSION THREAD

**Discussion by ghost** [area17/twill#1206] [Q&A] [answered]
Hi again!

Just wondering, is it possible to pass parameters to a repeater? 

For instance, I've got a few forms that have 'select' fields that have dynamic options based on what's in the DB, and I'd like to be able to reuse those forms for my repeaters e.g.

- Form
```
@extends('twill::layouts.form')

@section('fieldsets')
    @formFieldset(['id' => 'children', 'title' => 'Children'])
        @formField('repeater', ['type' => 'children'])
    @endformFieldset
@endsection
```

- Repeater
```
@twillRepeaterTitle('Child')
@twillRepeaterTrigger('Add child')
@twillRepeaterGroup('app')

 @formField('radios', [
    'name' => 'type_id',
    'label' => 'Type',
    'options' => $types,
    'inline' => true,
])
```

Basically, is there a way to pass a $types list to the repeater?

**Comment by pboivin** [comment, marked as answer]
Hi @aarontilt, this is a question that comes up regularly and will definitely be addressed in the near future. But for now, it's not possible to pass variables to repeater children from the form field. Depending on your use case, you may find `View::share()` useful. Fetching some data from the template is also possible:
```
 @formField('radios', [
    'name' => 'type_id',
    'label' => 'Type',
    'options' => \App\Models\Parent::getTypeOptions(),
    'inline' => true,
])
```
If you want to dig deeper, maybe this discussion can give you a bit more context: https://github.com/area17/twill/discussions/1079

**Reply by ghost** [reply]
Ah, that'll do for what I need. I'll take a look into the View::share() stuff too. Cheers!
